app/models/apple/episode.rb

```diff
@@ -58,23 +58,22 @@ def self.get_episodes(api, episodes)
       api.bridge_remote_and_retry!("getEpisodes", episodes.map(&:get_episode_bridge_params))
     end
 
+    # Creates the `#apple_sync_log` attributes for the given episodes
+    # and returns the created `SyncLog` records
+    # This indicates that there is a remote pair for the episode
     def self.poll_episode_state(api, show, episodes)
-      guid_to_apple_json = show.apple_episode_json.map do |ep_json|
-        [ep_json["attributes"]["guid"], ep_json]
-      end.to_h
-
       # Only sync episodes that have a remote pair
-      episodes_to_sync = episodes.filter { |ep| guid_to_apple_json[ep.guid].present? }
+      episodes_to_sync = episodes.filter { |ep| show.find_apple_episode_json_by_guid(ep.guid).present? }
 
       bridge_params = episodes_to_sync.map do |ep|
-        id = guid_to_apple_json[ep.guid]["id"]
-        guid = guid_to_apple_json[ep.guid]["attributes"]["guid"]
-        Episode.get_episode_bridge_params(api, id, guid)
+        apple_json = show.find_apple_episode_json_by_guid(ep.guid)
+        apple_episode_id = apple_json["id"]
+        Episode.get_episode_bridge_params(api, ep.feeder_id, apple_episode_id)
```

Comment:
Apple ID / GUID changes

This uses the new show level interface for looking up guids. This will handle the case where the episode has the apple episode id set as the feed item guid (in the RSS).

If the episode's item guid matches an Apple episode id, then join to the that.

At the end we upsert the "SyncLog", and subsequently we reference the remote episode only via it's `id` property.